viewer_clients/web-client/scripts/codecheckerviewer/BugViewer.js

```diff
@@ -462,49 +462,201 @@ function (declare, dom, style, on, query, Memory, Observable, topic, entities,
         this.editor.highlightBugPathEvent(item.bugPathEvent);
     },
 
+    getIconClass : function(item, opened) {
+      if (item == this.model.root) {
+        return (opened ? "dijitFolderOpened" : "dijitFolderClosed");
+      } else if (!item.isLeaf) {
+        switch (item.kind) {
+          case 'severity':
+            return "customIcon severity-" + item.id;
+          case 'bugpath':
+            return (opened ? "customIcon pathOpened" : "customIcon pathClosed");
+        }
+
+        return (opened ? "dijitFolderOpened" : "dijitFolderClosed");
+      } else if (item.isLeaf) {
+        switch (item.kind) {
+          case 'msg':
+            return "customIcon assume_msg nocolor";
+          case 'event':
+            return (item.iconOverride ? "customIcon " + item.iconOverride : "customIcon msg");
+          case 'result':
+            return "customIcon result";
+        }
+
+        return "dijitLeaf";
+      }
+    },
+
     _onNodeMouseEnter : function (node) {
-      if (node.item.isLeaf)
-        Tooltip.show(node.item.name, node.domNode, ['above']);
+      if (node.item.tooltip)
+        Tooltip.show(node.item.tooltip, node.domNode, ['above']);
     },
 
     _onNodeMouseLeave : function (node) {
       if (node.item.isLeaf)
         Tooltip.hide(node.domNode);
     },
 
+    _highlight_colours : [
+      "#ffffff",
+      "#e9dddd",
+      "#dde0eb",
+      "#e5e8e5",
+      "#cbc2b6",
+      "#b8ccea",
+      "#c1c9cd",
+      "#a7a28f"
+    ],
+
+    _highlightStep : function(stack, step) {
+      var that = this;
+      var msg = step.msg;
+
+      // The background must be saved BEFORE stack transition. Calling is in the caller, return is in the called func.
+      var highlight = {
+        background : stack.background,
+        iconOverride : undefined
+      };
+
+      var stackTransition = function() {
+        var contains = function(substr) {
+          return msg.indexOf(substr) !== -1;
+        };
+        var extractFuncName = function(prefix) {
+          if (contains(prefix)) {
+            return msg.replace(prefix, "").replace("'", "");
+          }
+          return false;
+        };
+
+        var func;
+        if (func = extractFuncName("Calling ")) {
+          stack.funcStack.push(func);
+          stack.background = that._highlight_colours[stack.funcStack.length % that._highlight_colours.length];
+
+          highlight.iconOverride = "calling";
+        } else if (contains("Entered call from ")) {
```

Comment:
No, it was an accidental oversight due to having multiple redesigns in the code. It will be fixed in the next commit, along with a slight README update.